### PR TITLE
Add localization support to WebTorrent

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/24555
Resolves https://github.com/brave/brave-browser/issues/11763

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
       "es2019",
       "esnext"
     ],
+    "resolveJsonModule": true,
```
Comment (fallaciousreasoning) [inline tsconfig.json:23]: @petemill do you think this is okay? I did it so I could get strong typings for the localization message keys.

Comment (petemill) [reply tsconfig.json:23]: I think so. Nice idea.

File: browser/ui/webui/brave_webui_source.cc [hunk 2]
```diff
@@ -1059,6 +1059,11 @@ void CustomizeWebUIHTMLSource(const std::string &name,
         { "confirmationNotice",
             IDS_BRAVE_WEBCOMPATREPORTER_CONFIRMATION_NOTICE },
       }
+    }, {
+      std::string("brave_webtorrent"), {
+        { "webtorrentPeer", IDS_WEBTORRENT_PEER },
+        { "webtorrentPeers", IDS_WEBTORRENT_PEERS },
+      }
```
Comment (petemill) [inline browser/ui/webui/brave_webui_source.cc:1066]: Where are these used? As far as I know, this area map is used for WebUI pages only and not extension. If it is needed, whilst this is probably ok, I think this technique (putting all the strings for all WebUIs in a single file (which is very long now) is deprecated. And would prefer if it went directly in the specific WebUI.

File: components/brave_webtorrent/BUILD.gn [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import("//brave/components/common/typescript.gni")
 import("//brave/resources/brave_grit.gni")
 import("//tools/grit/repack.gni")
+import("//build/config/locales.gni")
+import("//tools/grit/repack.gni")
```
Comment (petemill) [inline components/brave_webtorrent/BUILD.gn:5]: nit: prefer alphabetical sorting, I think

File: components/brave_webtorrent/extension/_locales/en_GB/messages.json [hunk 4]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (petemill) [inline components/brave_webtorrent/extension/_locales/en_GB/messages.json:1]: Is this different from the en_US file?

**Review by petemill — CHANGES_REQUESTED** [review]
Just a question or 2, as below.

Also, if we want actual translations to be done, I think we might have to add the new messages path to `build/commands/lib/l10nUtil.js` and then I'm not sure if anything needs to be done in transifex to set up a new project or something like that. Perhaps @mkarolin would know?

**Comment by mkarolin** [comment]
> l10nUtil.js

@fallaciousreasoning do we expect the strings to change often? If this is more of a one time translation deal rather than frequent updates then it's better to set it up similarly to how we localize greaselion. Just grep for `geaselion` in `brave/build/commands/lib/pushL10n.js|pullL10n.js` and under `brave/scripts/*.py` (and also for `site-specific-scripts`).

**Comment by bsclifton** [comment]
Closing as stale. I updated both issues to be a P5 since we haven't taken a look at this in 1.5 years

**Pull request closed without merging**